util/pkg/vfs/s3context.go

```diff
@@ -84,7 +82,7 @@ func (s *S3Context) getClient(region string) (*s3.S3, error) {
 		var err error
 		endpoint := os.Getenv("S3_ENDPOINT")
 		if endpoint == "" {
-			config = aws.NewConfig().WithRegion(region)
+			config = aws.NewConfig().WithRegion(region).WithUseDualStack(true)
```

Comment:
Do all regions support the dual stack endpoint for S3?

Reply:
The dual-stack S3 endpoints are missing in `cn-north-1` and `cn-northwest-1`.

They do exist in the GovCloud regions. I don't know the names of the isolated regions.

Reply:
As far as I can tell, yes. At least I haven't found a region where this doesn't work.

I can feed in `IPv6Only` here as well to be safe. That would at least put this behind a feature flag.

Reply:
We could require users in said regions to set the `S3_ENDPOINT` environment variable...

Reply:
Yeah. Come to think of it, making this code be aware of `IPv6Only` will be very tricky.

Reply:
Do we need a release note for folks in the China regions?

Reply:
Seems that we don't need to do anything for China regions:
https://aws.amazon.com/about-aws/whats-new/2020/05/amazon-s3-adds-support-ipv6-protocol-beijing-sinnet-ningxia-nwcd/

Reply:
Ah, different TLD.